### PR TITLE
Add a weather gadget

### PR DESCRIPTION
**Pull request by RealKC**
This will fail CI, but I'm opening it for some reviews

### CODE REVIEW

File: gadgets/weather/src/wttr.rs [hunk 1]
```diff
@@ -0,0 +1,90 @@
+use std::{fmt::Display, str::FromStr};
```
Comment (lucastarche) [inline gadgets/weather/src/wttr.rs:1]: This code looks familiar... :thonk:

:P

Comment (RealKC) [reply gadgets/weather/src/wttr.rs:1]: oh yeah, I took it from tarchboard, want me to link it somehow? (Or add you as a coauthor to the commit?)

**Review by lucastarche — APPROVED** [review]
Looks good! We might want to sideload asset loading (in the Background for instance) to a tokio task as well lator on though

File: gadgets/weather/src/wttr.rs (hunk 1 above)
Comment (lucastarche) [reply gadgets/weather/src/wttr.rs:1]: nah dwai

**Comment by RealKC** [comment]
> Looks good! We might want to sideload asset loading (in the Background for instance) to a tokio task as well lator on though

Yeah I'll eventually look into how to do that without blocking the tokio thread from doing network stuff (file I/O is synchronous on linux w/o using io_uring)

**Pull request merged**